tests/core/test_featurizer.py

```diff
@@ -1,111 +1,284 @@
-from rasa.core.featurizers import (
-    TrackerFeaturizer,
+import logging
+from typing import Text
+from unittest.mock import Mock
+import sys
+import asyncio
+
+from rasa.core.featurizers.tracker_featurizers import TrackerFeaturizer
+from rasa.core.featurizers.single_state_featurizer import (
     BinarySingleStateFeaturizer,
     LabelTokenizerSingleStateFeaturizer,
+    SingleStateFeaturizer,
 )
+import rasa.core
+from rasa.train import train_core, train_nlu, train
+from rasa.core.domain import Domain
 import numpy as np
+from rasa.nlu.constants import (
+    TEXT,
+    INTENT,
+    ACTION_NAME,
+    ACTION_TEXT,
+    ENTITIES,
+    FEATURE_TYPE_SENTENCE,
+)
+from rasa.core.constants import SLOTS, ACTIVE_LOOP
+import scipy.sparse
+from _pytest.monkeypatch import MonkeyPatch
+from pathlib import Path
+from tests.conftest import DEFAULT_CONFIG_PATH, DEFAULT_NLU_DATA
+from tests.core.conftest import (
+    DEFAULT_DOMAIN_PATH_WITH_SLOTS,
+    DEFAULT_STORIES_FILE,
+)
 
 
 def test_fail_to_load_non_existent_featurizer():
     assert TrackerFeaturizer.load("non_existent_class") is None
 
 
-def test_binary_featurizer_handles_on_non_existing_features():
-    f = BinarySingleStateFeaturizer()
-    f.input_state_map = {"a": 0, "b": 3, "c": 2, "d": 1}
-    f.num_features = len(f.input_state_map)
-    encoded = f.encode({"a": 1.0, "b": 1.0, "c": 0.0, "e": 1.0})
-    assert (encoded == np.array([1, 0, 0, 1])).all()
+def test_single_state_featurizer_correctly_encodes_state_without_interpreter():
```

Comment:
one unit test should always just test one scenario. Either use `pytest.mark.parametrize` or define a new test

Reply:
this applies to the other tests in this file as well